src/main/java/org/Olivia/EventDispatcher.java

```diff
@@ -0,0 +1,230 @@
+        String content = "";
+        for (CalendarEntry e : entries) {
+            content = content + e.toString() + "\n";
+        }
+        ui.cout((UiHandler.generateSection(content)));
+        return 200;
+    }
+
+    /**
+     * calls corresponding functions based on the input command
+     *
+     * @param input the array of string that contains the command and parameters
+     * @return status code (adapted from http, with exceptions such as status 0 represents exit)
+     */
+    public int dispatchCommand(String input) throws Exception {
+        if (input == null) {
+            throw new InvalidParameterException("command string array is not expected to be null, internal error");
+        }
+        if (input.length() == 0) {
+            throw new IllegalArgumentException("Sorry, command length cannot be zero");
+            //return 400; //Bad request
+        }
+        if (input.toLowerCase().equals("exit") || input.toLowerCase().equals("bye")) {
+            return 0; //NOT http status code, exit
+        }
+        if (input.toLowerCase().equals("list") || input.toLowerCase().equals("ls")) {
+            return list();
+        }
+
+        String[] splited_input = input.toLowerCase().split(" ");
+        if (splited_input.length == 0) {
+            throw new IllegalArgumentException("Sorry, I don't seem to understand you");
+        }
+        if (splited_input[0].equals("help")) {
+            return help();
+        }
+        if (splited_input[0].equals("mark") || splited_input[0].equals("unmark")) {
+            return markAsDoneUndone(input);
+        }
+        if (splited_input[0].equals("todo") || splited_input[0].equals("event") || splited_input[0].equals("deadline")) {
+            return addEntryToCalendar(input);
+        }
+        if (splited_input[0].equals("delete")) {
+            return delete(input);
+        }
+        if (splited_input[0].equals("find") && splited_input.length > 1) {
+            return find(input);
+        }
+        throw new IllegalArgumentException("Sorry, I don't seem to understand you");
+        //return 500; //not implemented
+    }
+
+    /**
+     * the infinite loop that listens to whatever place that the command is coming from (currently only support console)
+     * when the command is read, it will be handed over to the dispatch_command function to parse and execute
+     * after which the dispatch_command function will return a status code (adapted from http, details in respective comments)
+     * this function will print error msg (if any)
+     *
+     * @return status code
+     */
+    public int startWorking() {
+        ui.cout(ui.getGreeting());
+        try {
+            this.disk.syncFromFile(this.table);
+        }
+        catch (Exception e) {
+            ui.cout(e.getMessage() + "\n");
+            ui.cout("Error syncing with disk, starting with a fresh new table" + "\n");
+        }
```

Comment:
The try-catch statement should have the follow form: 
```suggestion
        try {
            this.disk.syncFromFile(this.table);
        } catch (Exception e) {
            ui.cout(e.getMessage() + "\n");
            ui.cout("Error syncing with disk, starting with a fresh new table" + "\n");
        }
```